nixos/modules/services/matrix/synapse.nix

```
@@ -219,6 +227,34 @@ in {
           '';
         };
 
+        path = mkOption {
+          type = types.nullOr types.path;
+          default = null;
+          description = ''
+            Unix domain socket path to bind this listener to.
+
+            ::: {.note}
+              This option is incompatible with {option}`bind_addresses`, {option}`port`, {option}`tls`
+              and also does not support the `metrics` and `manhole` listener {option}`type`.
+          '';
+        };
+
+        mode = mkOption {
+          type = types.nullOr (types.strMatching "^[0,2-7]{3,4}$");
+          default = if path != null then "1660" else null;
```

```suggestion
          default = if config.path != null then "660" else null;
```

This should correctly set `null` as the mode when no `path` is configured.